File: python/day1.py
with open('../data/day1.txt') as doc:
    data = doc.read().split('\n\n')

# elf with max calories
max_calories = max(map(lambda x: sum([int(z) for z in x.split('\n') if z]), data))


# top three elves
result = list(map(lambda x: sum([int(z) for z in x.split('\n') if z]), data))
result.sort(reverse=True)
top_three_sum = sum(result[:3])


# results
assert max_calories == 71124
assert top_three_sum == 204639

File: rust/src/day1.rs
use std::fs::read_to_string;

fn max_calories() {
    let data = read_to_string("../data/day1.txt").expect("file not found");
    let result: u32 = data
        .split("\n\n")
        .map(|x| {
            x.split("\n")
                .filter(|item| !item.is_empty())
                .map(|y| y.parse::<u32>().unwrap())
                .sum::<u32>()
        })
        .max()
        .unwrap();
    println!("{:?}", result);
    assert_eq!(result, 71124)
}

fn sum_top3() {
    let data = read_to_string("../data/day1.txt").expect("file not found");
    let mut result: Vec<u32> = data
        .split("\n\n")
        .map(|x| {
            x.split("\n")
                .filter(|item| !item.is_empty())
                .map(|y| y.parse::<u32>().unwrap())
                .sum::<u32>()
        })
        .collect::<Vec<u32>>();

    result.sort_by(|a, b| b.cmp(a)); // inverse sort
    result.truncate(3);  // get only the first three elements

    let top_three_sum = result.iter().sum::<u32>();

    println!("{:?}", top_three_sum);
    assert_eq!(top_three_sum, 204639)
}